### ISSUE TITLE
TODO list

### ISSUE THREAD

**Issue by lldelisle** [deeptools/pyGenomeTracks#119]
I put here what I would like to put in pygenometracks.
Close future:
- for each track put a `set_properties_defaults` method.
- put every duplicated code in utilities

Far future:
- add a linting tool for the ini file?
- update all helps to show all possible options for each track and make a full documentation
- speed up when a gtf is used, for example creating a database only for one chromosome (see PR I made).
- BAM track
- TAD classification track
- allow to plot differences/log2 ratio in the fly
- allow a log axis
- write an example for a 4C track
- check the story of x-axis bottom/top
- add the possibility to make background
- check the position of arcs/triangle in links tracks see #103 
- put examples of loop tracks
- check that the ylim is adjusted when plotting loop or triangle for the links track without overlay
- put the barcolorscale aligned below.

**Comment by lldelisle** [comment]
- new track type which would be horizontal line

**Comment by lldelisle** [comment]
- adjust the number of character per line dynamically (see #140 )

**Comment by DonStephano** [comment]
Hi,
is the issue with the arcs/links solved? I also suggest that the arcs should start/end in the middle of the region (maybe as an option). But I really like the improvements for blotting arcs, keep up the fantastic work!

**Comment by lldelisle** [comment]
Hi,
No sorry this is still the start of the first to the end of the second... You are right I should put an option and this would solve the issue...

**Comment by DonStephano** [comment]
No worries. It would be cool, for now I just manipulated my arc-file with the different start/end regions

**Comment by DonStephano** [comment]
It would be also very very cool to plot 4C-like plots with pygenometracks (implementation of chicPlotViewpoint) :) @joachimwolff

**Comment by joachimwolff** [comment]
It is on the list: https://github.com/deeptools/pyGenomeTracks/issues/234

**Comment by DonStephano** [comment]
perfect! Thx a lot
great work 👍

**Comment by lldelisle** [comment]
add as argument to pgt --plottingWidth to specify the width of the plotting part instead of the page width.
The conversion should be:
```
fig_width = args.plottingWidth / (DEFAULT_MARGINS['right'] - DEFAULT_MARGINS['left']) * (1 + 2 / 3 * 0.01) / width_ratios[1] / sum(width_ratios)

```

**Comment by lldelisle** [comment]
Support UTR/coding plot for bed8 and bed9.

**Comment by lldelisle** [comment]
@DonStephano the links is fixed in version 3.5 with the option `use_middle`.

**Comment by lldelisle** [comment]
From BCC2020, +1 for BAM format.

**Comment by DonStephano** [comment]
Hey @lldelisle 
thanks a lot! I will try it soon.
Additionally from a seminar where I presented pyGenomeTracks: +1 bam format :)
Thanks for the great work!

**Comment by lldelisle** [comment]
Add label of bed tracks below the regions

**Comment by GDelevoye** [comment]
Hello,

While this is just a workaround, remark that [bedtools bamtobed](https://bedtools.readthedocs.io/en/latest/content/tools/bamtobed.html) can be used to turn a .bam into a .bed file which, in turn, is handled correctly by pyGenomeTracks. This could be either a temporary solution, or maybe even the basis of a future PR to handle the .bam files. One cannot install easily bedtools from PyPi, but [unofficial python bindings exist](https://github.com/daler/pybedtools), and bedtools can be easily installed through [conda](https://anaconda.org/bioconda/bedtools)). That's probably not the best option for a clean PR, but this would just work

**Comment by lldelisle** [comment]
Thanks @GDelevoye .
My idea of a BAM track would handle some features supported by IGV: per base coverage (optionally with colors for mismatches), pair matching for PE data etc. But this is a lot of work.

**Comment by GDelevoye** [comment]
I agree, I thought the same a few days ago. This would really be a killer feature
